xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/NfTokenMint.java

```diff
@@ -0,0 +1,89 @@
+package org.xrpl.xrpl4j.model.transactions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.primitives.UnsignedLong;
+import org.immutables.value.Value;
+import org.xrpl.xrpl4j.model.flags.Flags;
+
+import java.util.Optional;
+
+/**
+ * The {@link NfTokenMint} transaction creates an NfToken object and adds it to the
+ * relevant NfTokenPage object of the minter(issuer). If the transaction is
+ * successful, the newly minted token will be owned by the minter(issuer) account
+ * specified by the transaction.
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableNfTokenMint.class)
+@JsonDeserialize(as = ImmutableNfTokenMint.class)
+public interface NfTokenMint extends Transaction {
+
+  /**
+   * Construct a builder for this class.
+   *
+   * @return An {@link ImmutableNfTokenMint.Builder}.
+   */
+  static ImmutableNfTokenMint.Builder builder() {
+    return ImmutableNfTokenMint.builder();
+  }
+
+  /**
+   * Indicates the taxon associated with this token. The taxon is generally a
+   * value chosen by the minter of the token and a given taxon may be used for
+   * multiple tokens. The implementation reserves taxon identifiers greater
+   * than or equal to 2147483648 (0x80000000).
+   *
+   * @return Taxon of type {@link int} associated with the Token.
+   */
+  @JsonProperty("TokenTaxon")
+  UnsignedLong tokenTaxon();
+
+  /**
+   * Indicates the account that should be the issuer of this token. This value
+   * is optional and should only be specified if the account executing the
+   * transaction is not the Issuer of the `NfToken` object. If it is
+   * present, the MintAccount field in the AccountRoot of the `Issuer`
+   * field must match the `Account`, otherwise the transaction will fail.
+   *
+   * @return An {@link Optional} field Issuer of type {@link Address}.
+   */
+  @JsonProperty("Issuer")
+  Optional<Address> issuer();
+
+  /**
+   * Specifies the fee charged by the issuer for secondary sales of the Token,
+   * if such sales are allowed. Valid values for this field are between 0 and
+   * 50000 inclusive, allowing transfer rates between 0.000% and 50.000% in
+   * increments of 0.001%. This field must NOT be present if the
+   * `tfTransferable` flag is not set.
+   *
+   * @return An {@link Optional} field TransferFee of type {@link CurrencyAmount}.
+   */
+  @JsonProperty("TransferFee")
+  Optional<CurrencyAmount> transferFee();
```

Comment:
I think `CurrencyAmount` is the wrong type for this. It seems like it is not an amount but rather a percentage. If so, we should create a class called `TransferFee` that wraps a `BigDecimal`. In that class, we should either round to the correct number of decimals or enforce no more than 3 decimal places and also enforce bounds (between 0% and 50%)

Reply:
It is an unsigned integer. ref: https://xrpl.org/nftokenmint.html#nftokenmint-fields

Reply:
Ah right. In that case, we should still create a class called `TransferFee` that wraps an `UnsignedInteger`. It should do the following:
1) Check the bounds of the integer (ie <= 9999) (also are the docs correct here? Greg's code says max is 50% in increments of 0.001% but docs say max is 99.99% in increments of 0.01%)
2) Have a static constructor like `public static TransferFee ofPercent(BigDecimal percent)` so that developers can construct using percentages.